locations/vector_file_spider.py

```diff
@@ -0,0 +1,56 @@
+from typing import Iterable
+
+from geopandas import read_file
+from scrapy import Spider
+from scrapy.http import Response
+
+from locations.dict_parser import DictParser
+from locations.items import Feature
+
+
+class VectorFileSpider(Spider):
+    """
+    A VectorFileSpider is a lightweight spider for extracting features from a
+    vector file format response returned from a specified `start_urls[0]`.
+
+    A wide variety of vector file formats are supported including:
+    1. ESRI Shapefile
+    2. GeoPackage
+    3. FlatGeobuf
+    4. GeoJSON (don't use with this spider, use JSONBlobSpider instead)
+
+    Archived versions of the above listed vector file formats are also
+    supported. For example, a ZIP archive of an ESRI Shapefile and associated
+    files (.shp, .dbf, .shx).
+
+    The details of file format support is available via the following
+    documentation for the underlying Python and system libraries this spider
+    type depends upon:
+    1. https://geopandas.org/en/stable/docs/reference/api/geopandas.read_file.html
+    2. https://pyogrio.readthedocs.io/en/latest/supported_formats.html
+    3. https://gdal.org/en/stable/drivers/vector/index.html
+
+    To use this spider, specify a URL with `start_urls`.
+
+    Override `pre_process_data` and/or `post_process_item` methods to modify
+    the fields of data extracted from the source vector file. These overriden
+    methods can be used extract additional non-standard fields which
+    DictParser doesn't automatically extract, or to clean up fields of data
+    that are incorrect or formatted incorrectly.
+    """
+
+    def parse(self, response: Response) -> Iterable[Feature]:
+        feature_collection = read_file(response.body).to_geo_dict()
+        for feature in feature_collection["features"]:
+            feature.update(feature.pop("properties"))
+            feature["geometry"]["coordinates"] = list(feature["geometry"]["coordinates"])
+            self.pre_process_data(feature)
+            item = DictParser.parse(feature)
```

Comment:
We support using the whole GeoJSON geometry on the item, so we can just copy that over instead of trying to detect it in DictParser:

```suggestion
            self.pre_process_data(feature)
            item = DictParser.parse(feature)
            item.geometry = feature["geometry"]
```

Reply:
Thanks for the catch.

I should have documented this at the time. geopandas.read_file returns coordinates as a tuple, not a list, as you'd typically find in GeoJSON that has been read from a file or is being written to a file.

I've fixed it in DictParser by allowing conversion from tuples to list so that internally ATP will always extract and work with coordinates as lists (not potentially two different types).

Reply:
Hold that. I've got to fix it up further on second inspection

Reply:
I've proposed improved DictParser geometry parsing so that GeoJSON Point positions as lists or tuples are equally parsed. No need for spiders to care anymore.